pkg/pipeline/stream/processor.go

```diff
@@ -42,40 +55,59 @@ func (n *ProcessorNode) Run(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-
 	defer cleanup()
+
+	n.logger.Debug(ctx).Msg("opening processor")
+	err = n.Processor.Open(ctx)
+	if err != nil {
+		n.logger.Err(ctx, err).Msg("failed opening processor")
+		return cerrors.Errorf("couldn't open processor: %w", err)
+	}
```

Comment:
If we hit this error the processor will still be running (or at least the processor instance will have the boolean `running` set to `true`). We need to either call `Teardown` here or in `RunnableProcessor.Open` on an error. I think calling it here would be a bit more explicit, this way `RunnableProcessor.Open` doesn't make any assumptions on how it's used.